gm4_heart_canisters/data/heart_canisters/functions/main.mcfunction

```diff
@@ -0,0 +1,9 @@
+# check how many heart canisters the player has
+execute as @a[gamemode=!spectator,gamemode=!creative,nbt={Inventory:[{tag:{gm4_heart_canister:1b}}]}] run function heart_canisters:check_heart_canister_count
+
+# give the player the health boost effect
+execute as @a[gamemode=!spectator,gamemode=!creative,scores={gm4_heart_can=1..}] run function heart_canisters:apply_health_boost
```

Comment:
I think it'd be smart to move this function call into the `#heart_canisters:check_heart_canister_count` function. That way it only runs if there are players with canisters around without the need for another check if anyone's using a heart can.

Reply:
The reason these two are separate is to make it easier in the future to make for example Zauber Armor compatible. All that would have to be done is add a function tag call in between the `check_heart_canister_count` and `apply_health_boost` calls. Zauber Armor would then modify the `gm4_heart_can` scoreboard.

Reply:
Note that zauber armor uses attributes to increase the health of the player, not the health boost effect. Heart cans *would* use that too if they didnt have to be worn or held for attributes to take effect.

Reply:
Ohh okay my bad